shared/src/main/scala/ast/codec/MamlCodecInstances.scala

```diff
@@ -68,54 +63,74 @@ trait MamlCodecInstances extends MamlUtilityCodecs {
     Encoder.forProduct3("args", "classifications", "symbol")(u => (u.children, u.classMap, u.sym))
 
   implicit lazy val decodeFocalMax: Decoder[FocalMax] =
-    Decoder.forProduct3("args", "neighborhood", "target")(FocalMax.apply)
+    Decoder.forProduct3[FocalMax, List[Expression], Neighborhood, Option[TargetCell]]("args", "neighborhood", "target") {
```

Comment:
We're defining codecs for all focal ops manually so I decided just to modify them. 
I think that TargetCell is not an optional field in all the focal ops => so it is only a parser logic to interpret it as a default field in a correct fashion. / cc @CloudNiner